src/managers/input/InputManager.cpp

```diff
@@ -421,7 +421,7 @@ void CInputManager::mouseMoveUnified(uint32_t time, bool refocus) {
             m_bLastFocusOnLS = false;
             return; // don't enter any new surfaces
         } else {
-            if (((FOLLOWMOUSE != 3 && allowKeyboardRefocus) && (*PMOUSEREFOCUS || m_pLastMouseFocus != pFoundWindow)) || refocus) {
+            if ((FOLLOWMOUSE != 3 && allowKeyboardRefocus) && ((*PMOUSEREFOCUS || m_pLastMouseFocus != pFoundWindow) || refocus)) {
```

Comment:
(not completely sure this is completely right (haven't tested with FOLLOWMOUSE = 3). Just wanted to make `refocus` respect `allowKeyboardRefocus`)

Reply:
why is this even in this MR? doesn't seem related

Reply:
I also don't understand the point here?

Reply:
I tweaked the logic so that this won't change how FOLLOWMOUSE = 3 behaves.

Reply:
The point is that we shouldn't refocus (change keyboard focus) when `allowKeyboardRefocus` is false (even on a click/force refocus), because there's a layer that has grabbed the keyboard.

Reply:
ok